plugins/plotly-express/src/Chart.tsx

```diff
@@ -0,0 +1,251 @@
+import React, { Component, ReactElement, RefObject } from 'react';
+import memoize from 'memoize-one';
+import { IconDefinition } from '@deephaven/icons';
+import Log from '@deephaven/log';
+import type { Layout, Icon, Data } from 'plotly.js';
+import Plotly from './plotly/Plotly';
+import Plot from './plotly/Plot';
+import PlotlyExpressChartModel from './PlotlyExpressChartModel';
+
+const log = Log.module('PlotlyExpressChart');
+
+interface ChartProps {
+  model: PlotlyExpressChartModel;
+  isActive: boolean;
+  onDisconnect: () => void;
+  onReconnect: () => void;
+  onUpdate: (obj: { isLoading: boolean }) => void;
+  onError: (error: Error) => void;
+}
+
+interface ChartState {
+  data: Partial<Data>[] | null;
+  layout: Partial<Layout>;
+  revision: number;
+}
+
+export class Chart extends Component<ChartProps, ChartState> {
```

Comment:
We should totally be able to use `Chart` from `@deephaven/chart`, no? Just the `ChartPanel` from `@deephaven/core-plugins` is harder. This is fine as is though as it works, but could share more code I think.

Reply:
No because we need the `Plot` to include all the plotly types. Right now community still just includes the base 5-6 we use there. `Chart` is what renders `Plot`

Could just extend `Chart` and change the render method possibly. Or make a future change to `Chart` to allow `Plot` as a prop or something

Reply:
And the 2d/3d chart buttons is different here than our existing chart

Reply:
And the 2d/3d chart buttons is different here than our existing chart